pyincore/analyses/mlenabledcgeslc/mlcgeslc.py

```diff
@@ -0,0 +1,173 @@
+            base_file_path,
+            GI_base_val_file,
+        ),
+        os.path.join(
+            base_file_path,
+            HH_base_val_file,
+        ),
+        os.path.join(
+            base_file_path,
+            FD_base_val_file,
+        ),
+        os.path.join(
+            base_file_path,
+            Base_KAP_file,
+        ),
+    ]
+
+    def __init__(self, incore_client: IncoreClient):
+        sectors, base_cap_factors, base_cap, model_coeffs, cap_shock_sectors = parse_files(self.model_filenames, self.filenames)
+        self.base_cap_factors = base_cap_factors
+        self.base_cap = base_cap
+        self.model_coeffs = model_coeffs
+        self.cap_shock_sectors = cap_shock_sectors
+        super(MlEnabledCgeSlc, self).__init__(incore_client, sectors, labor_groups=[f"L{gp}" for gp in range(1, 5)]) # 4 labor groups
+
+    def get_spec(self):
+        return {
+            'name': 'Salt-Lake-cge',
+            'description': 'CGE model for Salt Lake City.',
+            'input_parameters': [
+                {
+                    'id': 'result_name',
+                    'required': False,
+                    'description': 'Result CSV dataset name prefix',
+                    'type': str
+                }
+
+            ],
+            'input_datasets': [
+                {
+                    'id': 'sector_shocks',
+                    'required': True,
+                    'description': 'Aggregation of building functionality states to capital shocks per sector',
+                    'type': ['incore:capitalShocks']
+                }
+            ],
+            'output_datasets': [
+                {
+                    'id': 'domestic-supply',
+                    'parent_type': '',
+                    'description': 'CSV file of resulting domestic supply',
+                    'type': 'incore:Employment'
+                },
+                {
+                    'id': 'gross-income',
+                    'parent_type': '',
+                    'description': 'CSV file of resulting gross income',
+                    'type': 'incore:Employment'
+                },
+                {
+                    'id': 'pre-disaster-factor-demand',
+                    'parent_type': '',
+                    'description': 'CSV file of factor demand before disaster',
+                    'type': 'incore:FactorDemand'
+                },
+                {
+                    'id': 'post-disaster-factor-demand',
+                    'parent_type': '',
+                    'description': 'CSV file of resulting factor-demand',
+                    'type': 'incore:FactorDemand'
+                },
+                {
+                    'id': 'household-count',
+                    'parent_type': '',
+                    'description': 'CSV file of household count',
+                    'type': 'incore:HouseholdCount'
+                }
+            ]
+        }
+
+    def run_analysis(self) -> None:
```

Comment:
Why do you overrride the base analysis run_analysis method? Why not just implement this all in the "run" method?

Reply:
I think you could replace what you have above with:

def run(self):

That way the base analysis which checks the dataset input types doesn't get overridden.

Reply:
Yes, I didn't realize that. I have changed it now.